src/Consolonia.Core/Helpers/InputProcessing/InputProcessor.cs

```diff
@@ -0,0 +1,78 @@
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Consolonia.Core.Helpers.InputProcessing
+{
+    public class InputProcessor<T>(IEnumerable<IMatcher<T>> matchers)
+    {
+        private int _previousTopMatcherIndex = -1;
+        private ImmutableArray<IMatcher<T>> Matchers { get; } = [..matchers];
```

Comment:
man I just haven't absorbed this new syntax yet. I don't think of using it!

Reply:
I have no idea how it works. It's just resharper proposed to replace.